### PR TITLE
WT 7268 Fixes of coverity issues related to the test harness files

### PR DESCRIPTION
**Pull request by etienneptl**

### CODE REVIEW

File: test/cppsuite/test_harness/test.h [hunk 1]
```diff
@@ -127,6 +131,7 @@ class test {
         _thread_manager->join();
 
         /* Validation stage. */
+        testutil_check(_configuration->get_bool(ENABLE_TRACKING, enable_tracking));
```
Comment (luke-pearson) [inline test/cppsuite/test_harness/test.h:134]: Call is_enabled from the tracking component instead of pulling out the configuration again.

Comment (luke-pearson) [reply test/cppsuite/test_harness/test.h:134]: Add is_enabled to the component class seeing as we're using it in a number of places now.

File: test/cppsuite/tests/poc.cxx [hunk 2]
```diff
@@ -88,6 +87,12 @@ main(int argc, char *argv[])
     std::cout << "Configuration\t:" << cfg << std::endl;
     std::cout << "Trace level\t:" << trace_level << std::endl;
 
-    poc_test(cfg, trace_level).run();
+    try {
+        poc_test(cfg, trace_level).run();
+    } catch (const std::exception &e) {
```
Comment (luke-pearson) [inline test/cppsuite/tests/poc.cxx:92]: Instead of catching an exception here use testutil_die with a message so we're consistent.

**Comment by tetsuo-cpp** [comment]
@etienneptl I've spotted a few more issues from the past few days while triaging Coverity. Do you mind if I push a few changes into this branch so we can get them all at once?

**Comment by luke-pearson** [comment]
@tetsuo-cpp please do.

**Pull request closed without merging**